### PR TITLE
Feat/Skeleton

### PR DESCRIPTION
**Pull request by guifeliper**
# Description
This PR contains the Skeleton component. The Skeleton component is a content loader image, it was created only with CSS and no other external library. To see all the functionalities of the component, please start the Storybook with `yarn storybook` and you can control its arguments. 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Which Platforms Have Been Used?

- [x] Google Chrome (Web, width > 1024px)
- [ ] Google Chrome (iPad)
- [ ] Google Chrome (iPhone 6)
- [ ] Google Chrome (iPhone 5/SE)
- [x ] Safari (Web, width > 1024px)
- [ ] Firefox (Web, width > 1024px)
- [ ] Edge (Web, width > 1024px)

# Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings

### CODE REVIEW

File: src/components/Skeleton/Skeleton.tsx [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+  }
+  50% {
+    opacity: 0.4;
+  }
+  100% {
+    opacity: 1;
+  }
+`;
+
+const Root = styled.div<{ variant: string; width: number | undefined; height: number | undefined }>`
```
Comment (RabbitDoge) [inline src/components/Skeleton/Skeleton.tsx:26]: `width: number | undefined;` in TS is usually written `width?: number`

File: src/components/Skeleton/Skeleton.tsx [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+  min-height: 20px;
+  display: block;
+  background-color: #bdc2c4;
```
Comment (RabbitDoge) [inline src/components/Skeleton/Skeleton.tsx:29]: How this render on dark mode ? Maybe you should use `textDisabled` from the theme. To be confirmed.

File: src/components/Skeleton/Skeleton.tsx [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+  border-radius: 4px;
```
Comment (RabbitDoge) [inline src/components/Skeleton/Skeleton.tsx:30]: `radii.small` from theme

Comment (RabbitDoge) [reply src/components/Skeleton/Skeleton.tsx:30]: better define this as a ternary : 

```
border-radius: ${({ variant, theme }) => (variant === 'circle ? theme,radii.circle : theme,radii.small)};
```

and remove the line 33

**Review by RabbitDoge — CHANGES_REQUESTED** [review]
Just some values to pull from the theme file

File: src/components/Skeleton/Skeleton.tsx [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+    width: 150px;
+    animation: ${waves} 2s cubic-bezier(0.4, 0, 0.2, 1) infinite;
+  }
+`;
+
+const Skeleton: React.FC<SkeletonProps> = ({ width, height, variant = "rect", animation = "pulse" }) => {
+  return (
+    <>
+      {animation === "pulse" && <Pulse variant={variant} width={width} height={height} />}
+      {animation === "waves" && <Waves variant={variant} width={width} height={height} />}
```
Comment (hachiojidev) [inline src/components/Skeleton/Skeleton.tsx:63]: Can we use the constants in `types`, e.g...

```
animation === animation.PULSE
```

File: src/components/Skeleton/Skeleton.tsx [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+
+const Root = styled.div<{ variant: string; width: number | undefined; height: number | undefined }>`
+  min-height: 20px;
+  display: block;
+  background-color: #bdc2c4;
+  border-radius: 4px;
+  width: ${({ width }) => (width ? `${width}px` : "100%")};
+  height: ${({ height }) => (height ? `${height}px` : "100%")};
+  ${({ variant }) =>
+    variant === "circle" &&
```
Comment (hachiojidev) [inline src/components/Skeleton/Skeleton.tsx:34]: `variant.CIRCLE`

File: src/components/Skeleton/Skeleton.tsx (hunk 1 above)
Comment (hachiojidev) [inline src/components/Skeleton/Skeleton.tsx:26]: This variant type should be `Variant` right?

**Review by hachiojidev — CHANGES_REQUESTED** [review]

File: src/components/Skeleton/Skeleton.tsx (hunk 3 above)
Comment (guifeliper) [reply src/components/Skeleton/Skeleton.tsx:30]: Excellent @RabbitDoge  I didn't know the theme variables my fault. Fixing it!

File: src/components/Skeleton/Skeleton.tsx (hunk 1 above)
Comment (guifeliper) [reply src/components/Skeleton/Skeleton.tsx:26]: Agreed!

File: src/components/Skeleton/Skeleton.tsx (hunk 4 above)
Comment (guifeliper) [reply src/components/Skeleton/Skeleton.tsx:63]: I am going to update this and other const variables in the next commit. However, I am going to use as Uppercase so we now that we are talking about the const.

File: src/components/Skeleton/Skeleton.tsx (hunk 2 above)
Comment (guifeliper) [reply src/components/Skeleton/Skeleton.tsx:29]: It is not looking so nice, I am going to see the behaviour for dark mode

Comment (guifeliper) [reply src/components/Skeleton/Skeleton.tsx:29]: Definitely looks much better:

File: src/components/Skeleton/Skeleton.tsx (hunk 1 above)
Comment (guifeliper) [reply src/components/Skeleton/Skeleton.tsx]: I agree the type should be Variant from the SkeletonProps.

**Review by hachiojidev — APPROVED** [review]

**Review by RabbitDoge — APPROVED** [review]

**Comment by bread-cn** [comment]
Dev kool 👏

**Pull request merged**